### ISSUE TITLE
Error in t(do.call(sparseMatrix, do.call(rbind, lapply(seq_along(blist),  :    invalid 'lazy' to R_sparse_transpose()

### ISSUE THREAD

**Issue by kennyyeo13** [cafferychen777/MicrobiomeStat#11]
Hi Chen, 

I ran the codes below as per the tutorial. However I got this error "Error in t(do.call(sparseMatrix, do.call(rbind, lapply(seq_along(blist),  : invalid 'lazy' to R_sparse_transpose()"

data(peerj32.obj)
alpha_test_results <- generate_alpha_test_pair(
  data.obj = peerj32.obj,  
  alpha.obj = NULL,
  time.var = "time",
  alpha.name = c("shannon"),
  subject.var = "subject",
  group.var = "group",
  adj.vars = "sex"
)

**Error
Using table 'feature.tab' in 'data.obj'.
Using table 'feature.tab' in 'data.obj'.
Calculating shannon diversity...
Diversity calculations complete.
Using table 'meta.dat' in 'data.obj'.
Error in t(do.call(sparseMatrix, do.call(rbind, lapply(seq_along(blist),  : 
  invalid 'lazy' to R_sparse_transpose()**

Hope you can help

thanks
Kenny

**Comment by cafferychen777** [comment]
Hi @kennyyeo13 ,

Thank you for reaching out and sharing the error details with me.

I've run the provided code on my MacBook and also through devtools tests, and I didn't encounter the issue you described. Given the error message and the context, I suspect that the problem may stem from differences in R and package versions.

Here's my session info for reference:

```
R version 4.3.1 (2023-06-16)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.0

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Asia/Shanghai
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
[1] MicrobiomeStat_1.1.1 tibble_3.2.1         rlang_1.1.1         

loaded via a namespace (and not attached):
  [1] remotes_2.4.2.1     testthat_3.1.10     permute_0.9-7       magrittr_2.0.3      clue_0.3-64        
  [6] matrixStats_1.0.0   compiler_4.3.1      mgcv_1.8-42         systemfonts_1.0.4   callr_3.7.3        
 [11] vctrs_0.6.3         rmutil_1.1.10       stringr_1.5.0       profvis_0.3.8       pkgconfig_2.0.3    
 [16] crayon_1.5.2        fastmap_1.1.1       backports_1.4.1     ellipsis_0.3.2      labeling_0.4.3     
 [21] utf8_1.2.3          modeest_2.4.0       promises_1.2.1      rmarkdown_2.24      sessioninfo_1.2.2  
 [26] ps_1.7.5            nloptr_2.0.3        ragg_1.2.5          purrr_1.0.2         xfun_0.40          
 [31] cachem_1.0.8        later_1.3.1         ggprism_1.0.4       broom_1.0.5         parallel_4.3.1     
 [36] prettyunits_1.1.1   cluster_2.1.4       R6_2.5.1            stringi_1.7.12      RColorBrewer_1.1-3 
 [41] boot_1.3-28.1       pkgload_1.3.2.1     rpart_4.1.19        brio_1.1.3          numDeriv_2016.8-1.1
 [46] Rcpp_1.0.11         iterators_1.0.14    knitr_1.43          usethis_2.2.2       httpuv_1.6.11      
 [51] Matrix_1.6-1        splines_4.3.1       tidyselect_1.2.0    rstudioapi_0.15.0   yaml_2.3.7         
 [56] vegan_2.6-4         timeDate_4022.108   codetools_0.2-19    miniUI_0.1.1.1      processx_3.8.2     
 [61] pkgbuild_1.4.2      lattice_0.21-8      lmerTest_3.1-3      shiny_1.7.5         withr_2.5.0        
 [66] evaluate_0.21       stable_1.1.6        desc_1.4.2          urlchecker_1.0.1    pillar_1.9.0       
 [71] renv_1.0.2          foreach_1.5.2       generics_0.1.3      philentropy_0.7.0   rprojroot_2.0.3    
 [76] ggplot2_3.4.3       munsell_0.5.0       scales_1.2.1        minqa_1.2.5         timeSeries_4031.107
 [81] xtable_1.8-4        glue_1.6.2          statip_0.2.3        pheatmap_1.0.12     tools_4.3.1        
 [86] lme4_1.1-34         spatial_7.3-16      fBasics_4022.94     fs_1.6.3            grid_4.3.1         
 [91] ape_5.7-1           tidyr_1.3.0         devtools_2.4.5      colorspace_2.1-0    nlme_3.1-162       
 [96] cli_3.6.1           textshaping_0.3.6   fansi_1.0.4         dplyr_1.1.3         gtable_0.3.4       
[101] ggh4x_0.2.6         stabledist_0.7-1    digest_0.6.33       ggrepel_0.9.3       farver_2.1.1       
[106] htmlwidgets_1.6.2   memoise_2.0.1       htmltools_0.5.6     lifecycle_1.0.3     statmod_1.5.0      
[111] mime_0.12           GUniFrac_1.7        MASS_7.3-60   
```

I recommend the following:

1. Please update your R to version 4.3.1.
2. Upgrade all the relevant packages to the versions that I'm using, especially `dplyr` and any other packages that might be related to the operation you're attempting.
3. Try running the code again after these updates.

Do let me know if the issue persists after these updates, and I'll be happy to assist you further.

Best regards,
Chen YANG

**Comment by kennyyeo13** [comment]
Thanks Chen, 
I'll check and reopen if there's any issues.

Cheers
Kenny

**Comment by sdoms** [comment]
This error comes from some incompatibility of the phyloseq and the lme4 package. What solves it for me is to restart the R session and not have phyloseq loaded or force install the phyloseq version directly from GitHub.

**Comment by cafferychen777** [comment]
Dear @sdoms ,

Thank you immensely for pointing out the potential incompatibility between the `phyloseq` and `lme4` packages. Your input is highly valuable to us, and it helps ensure that our package remains robust and user-friendly.

1. I wanted to let you know that I've made some updates to the `generate_alpha_test_pair` function. Specifically, I've removed the reliance on the `lme4` package and have now fully transitioned to using `lmerTest`. The updated `generate_alpha_test_pair` should be more optimized in terms of error handling and statistical analysis. I encourage you to use this updated version.

2. Regarding the `phyloseq` package, given that our package only utilizes it in the `mStat_convert_phyloseq_to_data_obj` function, I have decided to remove its dependency altogether. This step is crucial to avoid any potential conflicts and to maintain the package's stability.

Again, thank you for your invaluable feedback. Community insights like yours play an essential role in improving and refining our tools.

Best regards,

Chen YANG
